core/dbt/cli/flags.py

```diff
@@ -89,102 +69,121 @@ def args_to_context(args: List[str]) -> Context:
-                            raise BadOptionUsage(
-                                param_name,
-                                "Deprecated parameters can only be set via environment variables",
-                            )
-
-                        # rename for clarity
-                        dep_name = param_name
-                        new_name = DEPRECATED_PARAMS.get(dep_name)
-
-                        # find param objects for their envvar name
-                        try:
-                            dep_opt = [x for x in ctx.command.params if x.name == dep_name][0]
-                            new_opt = [x for x in ctx.command.params if x.name == new_name][0]
-                        except IndexError:
-                            raise Exception(
-                                f"No deprecated param name match from {dep_name} to {new_name}"
-                            )
-
-                        # remove param from defaulted set since the deprecated
-                        # value is not set from default, but from an env var
-                        if new_name in params_assigned_from_default:
-                            params_assigned_from_default.remove(new_name)
-
-                        # adding the deprecation warning function to the set
-                        deprecated_env_vars[new_name] = renamed_env_var(
-                            old_name=dep_opt.envvar,
-                            new_name=new_opt.envvar,
+                            f"No deprecated param name match from {dep_name} to {new_name}"
                         )
 
-                        object.__setattr__(self, new_name.upper(), param_value)
-                    else:
-                        object.__setattr__(self, param_name.upper(), param_value)
-                        if ctx.get_parameter_source(param_name) == ParameterSource.DEFAULT:
-                            params_assigned_from_default.add(param_name)
+                    # Remove param from defaulted set since the deprecated
+                    # value is not set from default, but from an env var.
+                    if new_name in params_assigned_from_default:
+                        params_assigned_from_default.remove(new_name)
+
+                    # Add the deprecation warning function to the set.
+                    assert isinstance(dep_param.envvar, str)
+                    assert isinstance(new_param.envvar, str)
+                    deprecated_env_vars[new_name] = renamed_env_var(
+                        old_name=dep_param.envvar,
+                        new_name=new_param.envvar,
+                    )
+
+                # Set the flag value.
+                is_duplicate = hasattr(self, param_name.upper())
+                is_default = (
+                    True
+                    if ctx.get_parameter_source(param_name) == ParameterSource.DEFAULT
+                    else False
+                )
```

Comment:
This can be simplified
```suggestion
                is_default = ctx.get_parameter_source(param_name) == ParameterSource.DEFAULT
```

Reply:
Good catch! Fixed.